src/easyscience/io/serializer_component.py

```diff
@@ -0,0 +1,81 @@
+#  SPDX-FileCopyrightText: 2023 EasyScience contributors  <core@easyscience.software>
+#  SPDX-License-Identifier: BSD-3-Clause
+#  © 2021-2023 Contributors to the EasyScience project <https://github.com/easyScience/EasyScience
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from .serializer_dict import SerializerDict
+
+if TYPE_CHECKING:
+    from .serializer_base import SerializerBase
+
+
+class SerializerComponent:
+    """
+    This base class adds the capability of saving and loading (encoding/decoding, serializing/deserializing) easyscience 
+    objects via the `encode` and `decode` methods. 
+    The default encoder is `SerializerDict`, which converts the object to a dictionary.
+
+    Shortcuts for dictionary and encoding is also present.
+    """
+
+    _CORE = True
```

Comment:
probably out of scope, since you just moved the file, but this isn't used anywhere so likely can be removed